extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js

```diff
@@ -119,18 +127,32 @@ class AmpFlyingCarpet extends AMP.BaseElement {
     );
   }
 
+  /** @override */
   layoutCallback() {
     try {
-      this.assertPosition();
+      this.assertPosition_();
     } catch (e) {
       // Collapse the element if the effect is broken by the viewport location.
       this./*OK*/collapse();
       throw e;
     }
     this.scheduleLayout(this.children_);
+    listen(this.element, 'amp:built', this.layoutBuiltChild_.bind(this));
```

Comment:
> I think we should listen to 'amp:built' from each child here. 

That would register `n` callbacks, when only one is needed.

> As this.element is promised to have been built when we call #layoutCallback.

I'm counting on that. That way, I know the `target` of the event is never this element, so it must be one of the children that just built.

> Also I think '#listenOnce` is more appropriate as an element will only be built once.

I would like this, but I can't reconcile it with the first point.